src/client/components/PureComponent/index.js

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable */
+
+import React from 'react'
+import _ from 'lodash'
+
+class PureComponent extends React.PureComponent {
+  shouldComponentUpdate(nextProps: Object, nextState: Object) {
+    return !_.isEqual(this.props, nextProps) || !_.isEqual(this.state, nextState)
+  }
+}
+
+export const pureComponent = BaseComponent => class extends PureComponent {
```

Comment:
Is it really sound to wrap functional components in a class? I lived with the assumption that these were lighter somehow.